File: 766 - Toeplitz Matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix: list[list[int]]) -> bool:
        row = len(matrix)
        col = len(matrix[0])
        for c in range(col):
            if not self.checkDiag(matrix, 0, c):
                return False
        for r in range(row):
            if not self.checkDiag(matrix, r, 0):
                return False
        return True
    def checkDiag(self, matrix, row, col):
        test = matrix[row][col]
        while row + 1 < len(matrix) and col + 1 < len(matrix[0]):
            row += 1
            col += 1
            if test != matrix[row][col]:
                return False
        return True

File: 766 - Toeplitz Matrix.java
class Solution {
    public boolean isToeplitzMatrix(int[][] matrix) {
        for (int i = 0; i < matrix[0].length; i++) {
            if (!checkDiag(matrix, 0, i)) {
                return false;
            }
        }
        for (int i = 1; i < matrix.length; i++) {
            if (!checkDiag(matrix, i, 0)) {
                return false;
            }
        }
        return true;
    }
    public boolean checkDiag(int[][] matrix, int row, int col) {
        int test = matrix[row][col];
        while (row + 1 < matrix.length && col + 1 < matrix[0].length) {
            row++;
            col++;
            if (test != matrix[row][col]) {
                return false;
            }
        }
        return true;
    }
}